api/oc_rep_encode_crc_internal.h

```diff
@@ -0,0 +1,46 @@
+/****************************************************************************
+ *
+ * Copyright (c) 2023 plgd.dev s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"),
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#ifndef OC_REP_ENCODE_CRC_INTERNAL_H
+#define OC_REP_ENCODE_CRC_INTERNAL_H
+
+#include "util/oc_features.h"
+
+#ifdef OC_HAS_FEATURE_CRC_ENCODER
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "api/oc_rep_encode_internal.h"
+
+#define OC_CRC_REP_NULL (0x0)
+
+#define OC_CRC_REP_TRUE (0x1)
+#define OC_CRC_REP_FALSE (0x0)
```

Comment:
this value need to be differ than `OC_CRC_REP_NULL` -> for each special type.

Reply:
Do we need a unique representations though? You have many values of different times that will have the same CRC result:

For example doubles and int64_t with the same bit pattern:

0xABCDEFGH is equal to some integer, some double and some uint8_t* byte array / string, all these different values map to the same bit pattern and therefore will have the same CRC value.

Maybe we should add one byte representing the type of the value to each CRC calculation?